src/test/java/framework/HerokuLandingPage.java

```diff
@@ -0,0 +1,113 @@
+package framework;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.testng.Assert;
+import ru.yandex.qatools.allure.annotations.Step;
+
+public class HerokuLandingPage extends BasePage {
+
+    private By topBar = By.xpath("//input[@id='aa-search-input']");
+    private By autocomplete = By.xpath("//span[@id='algolia-autocomplete-listbox-0']");
+    private By gigElements = By.xpath("//body/div[@class='container section']/div[@class='row']//div[@class='gig-card']");
+    private By priceElement = By.xpath("//div[@class='gig-card']//h3");
+    private By postTitle = By.xpath("//div[@class='gig-card']//h4");
+    private By postImage = By.xpath("//div[@class='row']//img");
+    private By joinButton = By.xpath("//a[@class='btn btn-success']");
+    private By nameFieldReg = By.xpath("//input[@name='username']");
+    private By emailFieldReg = By.xpath("//input[@name='email']");
+    private By passwordFieldReg = By.xpath("//input[@name='password']");
+    private By submitButtonReg = By.xpath("//button[@type='submit']");
+    private By alert = By.xpath("//div[@class='alert alert-success alert-dismissible']");
+    private By signInButton = By.xpath("//ul[@class='nav navbar-nav navbar-right']//li[1]");
+    private By emailFieldLogIn = By.xpath("//input[@id='email']");
+    private By passwordFieldLogIn = By.xpath("//input[@id='password']");
+    private By submitButtonLogIn = By.xpath("//button[@type='submit']");
+    private By userIcon = By.xpath("//body/nav[@class='navbar navbar-default navbar-static-top']/div[@class='container']/div[@id='bs-example-navbar-collapse-1']/ul[@class='nav navbar-nav navbar-right']/li[3]/a[1]");
+    private By logOutButton = By.xpath("//ul[@class='dropdown-menu']//li[3]");
+
+
+
+    HerokuActions herokuActions = new HerokuActions();
```

Comment:
all of heroku actions needs to be on basepage, I dont wanna see any object creation on page object other than field variables for locators